core/src/main/java/io/grpc/MethodDescriptor.java

```diff
@@ -57,15 +57,16 @@
   private final boolean idempotent;
   private final boolean safe;
 
-  private final AtomicReferenceArray<Object> rawMethodNames =
-      new AtomicReferenceArray<Object>(InternalKnownTransport.values().length);
+  // Must be set to InternalKnownTransport.values().length
+  // Not referenced to break the dependency.
+  private final AtomicReferenceArray<Object> rawMethodNames = new AtomicReferenceArray<Object>(1);
 
-  final Object getRawMethodName(InternalKnownTransport t) {
-    return rawMethodNames.get(t.ordinal());
+  final Object getRawMethodName(int transportOrdinal) {
```

Comment:
Though not related to this PR, this and the following method could use some javadoc.
The field rawMethodNames could also have a comment for how it works.
They are not obvious by looking at this file.